e2e/helpers/tableColumnSchemaEditor.ts

```diff
@@ -0,0 +1,294 @@
+import { Locator, Page, expect, test } from '@playwright/test'
+
+// Types -----------------------------------------------------------------------
+type FacetType = 'Range' | 'Values'
+export type ColumnSchemaConfig = {
+  name: string
+  type: string
+  size?: number
+  maxListLength?: number
+  defaultValue?: string
+  restrictValues?: string[]
+  facet?: FacetType
+}
+type ColumnSchemaKey = keyof ColumnSchemaConfig
+const columnNameToSchemaKey: Record<string, ColumnSchemaKey | string> = {
+  'Column ID': 'id',
+  'Column Name': 'name',
+  'Column Type': 'type',
+  Size: 'size',
+  'Max List Length': 'maxListLength',
+  'Default Value': 'defaultValue',
+  'Restrict Values': 'restrictValues',
+  Facet: 'facet',
+}
+
+// Set Individual Column Schema ------------------------------------------------
+const enterColumnName = async (
+  page: Page,
+  columnIndex: number,
+  columnName: string,
+) => {
+  await test.step('enter column name', async () => {
+    const columnNameFields = page.getByPlaceholder('Column name')
+    expect(columnNameFields).toHaveCount(columnIndex + 1)
+    await columnNameFields.nth(columnIndex).fill(columnName)
+  })
+}
+
+const selectColumnType = async (
+  page: Page,
+  columnIndex: number,
+  columnType: string,
+) => {
+  await test.step('select column type', async () => {
+    const columnTypeFields = page.getByLabel('Column Type')
+    await columnTypeFields.nth(columnIndex).click()
+    await page.getByRole('option', { name: columnType, exact: true }).click()
+  })
+}
+
+const setColumnMaxSize = async (
+  page: Page,
+  columnIndex: number,
+  maxSize: number,
+) => {
+  await test.step('set column size', async () => {
+    const columnMaxSizeFields = page.getByLabel('Maximum Size')
+    await columnMaxSizeFields.nth(columnIndex).fill(maxSize.toString())
+  })
+}
+
+const setColumnMaxListLength = async (
+  page: Page,
+  columnIndex: number,
+  maxListLength: number,
+) => {
+  await test.step('set column max list length', async () => {
+    const columnMaxListLengthFields = page.getByLabel('Maximum List Length')
+    await columnMaxListLengthFields
+      .nth(columnIndex)
+      .fill(maxListLength.toString())
+  })
+}
+
+const setColumnDefaultValue = async (
+  page: Page,
+  columnIndex: number,
+  defaultValue: string,
+  columnType: string,
+) => {
+  await test.step('set column default value', async () => {
+    const columnDefaultValueField = page
+      .getByLabel('Default Value')
+      .nth(columnIndex)
+    if (columnType === 'Boolean') {
+      await columnDefaultValueField.click()
+      await page.getByRole('option', { name: defaultValue }).click()
```

Comment:
`Boolean` fields provide a drop-down for default values, rather than an input field